render/query.go

```diff
@@ -239,7 +251,7 @@ func (r *Reply) getDataAggregated(ctx context.Context, cfg *config.Config, tf Ti
 
 	// Grouping metrics by aggregation steps and functions
 	var step int64
-	for _, m := range metricList {
+	for n, m := range metricListRuleLookup {
 		newStep, agg := targets.rollupObj.Lookup(m, uint32(age))
 		step = r.cStep.calculateUnsafe(step, int64(newStep))
 		if mm, ok := bodyAggregation[agg.Name()]; ok {
```

Comment:
258 + 260 are broken now. There must be metricList elements.

Reply:
Why ? It's a start point of this PR.  I pass reverse or unreverse metric name to rule lookup based on table config settings.
```
if targets.isReverse && targets.rollupUseReverted {
	metricListRuleLookup = metricListUnreverse
} else {
	metricListRuleLookup = metricList
}
```

Reply:
I talk about `m` in lines 258 + 260 specifically. Instead, there should be `metricList[n]`

Reply:
As a reference, see https://github.com/lomik/graphite-clickhouse/pull/143/files#diff-dd0047c86eb3449b47b84a87cbc1cf2bbbb726459e13916af2a1586abcea4399R414, it stays unchanged, and bodyAggregation should use metricList as it was before